script.js

```diff
@@ -0,0 +1,44 @@
+let saladMenuFileData = [], prices = [];
+let bill = 0, divContent="<table>";
+
+fetch("/salads.json")
+  .then((response) => response.json())
+  .then((saladMenuFileData) => createMenu(saladMenuFileData));
+
+  function setPrice(counter,saladMenuFileData){
+    if((counter + 1) % 3 == 0) return 15;
+    else if(saladMenuFileData[counter][0] == 'E' ||
+            saladMenuFileData[counter][0] == 'R' ||
+            saladMenuFileData[counter][0] == 'G' ) return 10;
+    else return 20;
+  }
+
+function createMenu(saladMenuFileData){
+  let divMenu = document.getElementById("menu");
+    for(let indexCounter=0; indexCounter < saladMenuFileData.length; indexCounter++){
+      prices[indexCounter] = setPrice(indexCounter, saladMenuFileData);
+       divContent += "<tr  >";
+       divContent += "<td class='classContentRow' id=salad" + indexCounter + ">" + (indexCounter + 1);
+       divContent += ". " + saladMenuFileData[indexCounter] +"</td>";
+       divContent += "<td> " + prices[indexCounter] + "zł</td>";
+       divContent += "<td><input type='checkbox' id='checkbox" + indexCounter + "' value='" + indexCounter + "'onclick='checkMenuBox(id,value)'></td>";
+       divContent += "</tr>";
+    }
+  divContent+="</table>";
+  divMenu.innerHTML = divContent;

```

Comment:
Zamiast tworzyć kontent w ten sposób mógłbyś wydzielić tą logikę do funkcji. Będzie się estetyczniej i przyjemniej czytać. Druga rzecz to parsowanie string na html za pomocą innerHTML. W tym przypadku nie jest to problem, ale w przypadku systemów CMS bądź jakichkolwiek innych gdzie treść wprowadza użytkownik można wstrzyknąć złośliwy skrypt JS i w ten sposób wejść w posiadanie jakichś wrażliwych informacji. Żeby tego uniknąć można skorzystać z sanitize. Czyli dekodować html treści wprowadzonych przez użytkownika. To jako ciekawostkę podaje. Jeżeli chcesz poczytać o tym to szukaj w google XSS (cross site scripting). A do tworzenia html korzystaj z funkcji createElement, appendChild, textNode itp.
https://www.tutorialspoint.com/how-to-add-a-new-element-to-html-dom-in-javascript